src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -2520,23 +2520,12 @@ class StubGenerator: public StubCodeGenerator {
     assert((base_offset2 % (UseCompactObjectHeaders ? 4 :
                             (UseCompressedClassPointers ? 8 : 4))) == 0, "Must be");
 
-    // cnt2 == amount of characters left to compare
-    // Check already loaded first 4 symbols
-    __ inflate_lo32(tmp3, isLU ? tmp1 : tmp2);
-    __ mv(isLU ? tmp1 : tmp2, tmp3);
-    __ addi(str1, str1, isLU ? wordSize / 2 : wordSize);
-    __ addi(str2, str2, isLU ? wordSize : wordSize / 2);
-    __ subi(cnt2, cnt2, wordSize / 2); // Already loaded 4 symbols
-
-    __ xorr(tmp3, tmp1, tmp2);
-    __ bnez(tmp3, CALCULATE_DIFFERENCE);
-
     Register strU = isLU ? str2 : str1,
              strL = isLU ? str1 : str2,
              tmpU = isLU ? tmp2 : tmp1, // where to keep U for comparison
              tmpL = isLU ? tmp1 : tmp2; // where to keep L for comparison
 
-    if (AvoidUnalignedAccesses && (base_offset1 % 8) == 0) {
+    if (AvoidUnalignedAccesses && (base_offset1 % 8) != 0) {
```

Comment:
I find that the similar check is in `C2_MacroAssembler::string_compare` for the UU/LL cases [1].
Seems more consistent if we move it into the counterpart `generate_compare_long_string_same_encoding`.

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp#L1443

Reply:
Agree, we should refactor the code a bit to make it more readable.
As it seems just a refactor, so I can do it in another pr, how do you think about it?
At the same time I can also clean the invocation of `compare_string_16_bytes_same` from `generate_compare_long_string_same_encoding`, I don't like the implicit registers passing between them.

Reply:
Sure. Seems the LL/UU cases are kind of different as they already emit direct 8-byte loads before the stub. So not sure if it's doable to move the check.